rosbag2/src/rosbag2/writer.cpp

```diff
@@ -63,7 +69,14 @@ void Writer::open(
   if (!storage_) {
     throw std::runtime_error("No storage could be initialized. Abort");
   }
+
   uri_ = storage_options.uri;
+
+  metadata_ = rosbag2_storage::BagMetadata{};
+  metadata_.storage_identifier = storage_->get_storage_identifier();
+  metadata_.starting_time = std::chrono::time_point<std::chrono::high_resolution_clock>(
+    std::chrono::nanoseconds::max());
+  metadata_.relative_file_paths.push_back(storage_->get_relative_path());
```

Comment:
```suggestion
  metadata_.relative_file_paths = {storage_->get_relative_path()};
```

should work and would make it more explicit we are actually resetting the vector

Reply:
http://cpp.sh/7j2mn

Reply:
Applying suggestion.